dbrp/bucket_service.go

```diff
@@ -37,13 +38,102 @@ func (s *BucketService) DeleteBucket(ctx context.Context, id platform.ID) error
 		BucketID: &bucket.ID,
 	})
 	if err != nil {
-		logger.Error("Failed to lookup DBRP mappings for Bucket.", zap.Error(err))
+		logger.Debug("Failed to lookup DBRP mappings for bucket", zap.Error(err))
 		return nil
 	}
 	for _, m := range mappings {
 		if err := s.DBRPMappingService.Delete(ctx, bucket.OrgID, m.ID); err != nil {
-			logger.Error("Failed to delete DBRP mapping for Bucket.", zap.Error(err))
+			logger.Debug("Failed to delete DBRP mapping for bucket", zap.Error(err))
 		}
 	}
 	return nil
 }
+
+// ParseDBRP parses DB and RP strings out of a bucket name
+func ParseDBRP(bucketName string) (string, string) {
+	db, rp, isCut := strings.Cut(bucketName, "/")
+	if isCut {
+		return db, rp
+	}
+	return bucketName, "autogen"
+}
+
+// CreateBucket creates a new bucket and sets b.ID with the new identifier.
+// Also creates a corresponding auto-DBRP mapping.
```

Comment:
Do we need an explicitly created DBRP mapping if we can automate it?

I was thinking that for a db/rp pair, to find the bucket we would:
* first look at the table of mappings to see if there is a bucket there
* if not, see if there is a name match a la `ParseDBRP` above
* if not, 404

Reply:
If we do that we don't need to do much migration / extra creation.

Reply:
And it saves us from 'oh I deleted my dbrp mapping' tripping people up.

Reply:
That probably would've been a better route to go, but I saw I could override the DBRP bucket service wrapper implementations, and knew I could create and update DBRPs there.
Do you think it's now worth reimplementing DBRPs virtually instead?

The Create, Update, and Delete bucket methods all handling DBRP mapping changes now makes it seem like a user wouldn't really need to bother with creating or deleting DBRPs manually either...